tests/test_utils.py

```diff
@@ -3,15 +3,16 @@
 
 import scrapy
 from scrapy.http import TextResponse
+from scrapy.settings import Settings
```

Comment:
It would be nice to have a test that checks that overriding on providers works. Something like having two providers with some overlapping class, and then provide them in a different order and be sure that the response comes from the proper provider.